http/preview.go

```diff
@@ -0,0 +1,98 @@
+package http
+
+import (
+	"fmt"
+	"image"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/disintegration/imaging"
+	"github.com/gorilla/mux"
+
+	"github.com/filebrowser/filebrowser/v2/files"
+)
+
+const (
+	sizeThumb = "thumb"
+	sizeBig   = "big"
+)
+
+type imageProcessor func(src image.Image) (image.Image, error)
+
+var previewHandler = withUser(func(w http.ResponseWriter, r *http.Request, d *data) (int, error) {
+	if !d.user.Perm.Download {
+		return http.StatusAccepted, nil
+	}
+	vars := mux.Vars(r)
+	size := vars["size"]
+	if size != sizeBig && size != sizeThumb {
+		return http.StatusNotImplemented, nil
+	}
+
+	// Resolve file path from URL
+	path := "/" + strings.Join(strings.Split(r.URL.Path, "/")[2:], "/")
```

Comment:
Why not to use the path as a variable like the size? Now it looks like a real magic :)

Reply:
Do you mean like this?
```
api.PathPrefix("/preview/{size}/{path}")
```

Reply:
Yes

Reply:
This way doesn't work when the path like `photos/xxx.jpg`,the request URI is `/preview/big/photos/xxx.jpg`.

Reply:
Use regexp for that `/preview/{size}{path:.*}`

Reply:
I got it👍